lib/system/seqs_v2.nim

```diff
@@ -81,7 +81,8 @@ proc shrink*[T](x: var seq[T]; newLen: Natural) {.tags: [], raises: [].} =
     #sysAssert newLen <= x.len, "invalid newLen parameter for 'shrink'"
     when not supportsCopyMem(T):
       for i in countdown(x.len - 1, newLen):
-        reset x[i]
+        {.cast(raises: []), cast(tags: []).}:
```

Comment:
Do this in `reset` itself?

Reply:
Yeah, will do it after I learn how many effects on important packages since custom destructors might have exceptions.